File: jarboot-ui2/src/services/UserService.ts
import Request from '@/common/Request';
import type { ResponseVo } from '@/types';

const urlBase = '/api/jarboot/user';

/**
 * 用户管理操作
 */
export default class UserService {
  public static createUser(username: string, fullName: string, password: string, roles: string, userDir: string, avatar: string | null = null) {
    const form: FormData = new FormData();
    form.append('username', username);
    form.append('fullName', fullName);
    form.append('password', password);
    form.append('roles', roles);
    form.append('userDir', userDir || '');
    avatar && form.append('avatar', avatar);
    return Request.post<ResponseVo>(urlBase, form);
  }

  public static updateUser(
    username: string,
    fullName: string | null,
    roles: string | null,
    userDir: string | null,
    avatar: string | null = null
  ) {
    const form: FormData = new FormData();
    form.append('username', username);
    form.append('fullName', fullName ?? '');
    form.append('roles', roles ?? '');
    form.append('userDir', userDir ?? '');
    avatar && form.append('avatar', avatar);
    return Request.post<ResponseVo>(urlBase + '/update', form);
  }

  public static deleteUser(id: number) {
    const form: FormData = new FormData();
    form.append('id', '' + id);
    return Request.delete<ResponseVo>(urlBase, form);
  }

  public static updateUserPassword(username: string, password: string, oldPassword: string) {
    const form: FormData = new FormData();
    form.append('username', username);
    form.append('password', password);
    form.append('oldPassword', oldPassword || '');
    return Request.put<ResponseVo>(urlBase, form);
  }

  public static getUsers(username: string, role: string, pageNo: number, pageSize: number) {
    return Request.get(`${urlBase}/getUsers`, { username, role, pageNo, pageSize });
  }

  public static getAvatar(username: string) {
    return Request.get<string>(`${urlBase}/avatar`, { username });
  }

  public static getUserDirs() {
    return Request.get<string[]>(`${urlBase}/userDirs`, {});
  }

  public static createOpenApiToken(username?: string, password?: string, expire?: number) {
    const form: FormData = new FormData();
    if (username && username.length > 0) {
      form.append('username', username);
    }
    if (password && password.length > 0) {
      form.append('password', password);
    }
    if (expire && expire > 0) {
      form.append('expireTimestamp', expire + '');
    }
    return Request.post<any>(`${urlBase}/open-api-token`, form);
  }

  public static getTokens(username: string, pageNo: number, pageSize: number) {
    return Request.get(`${urlBase}/open-api-token`, { username, pageNo, pageSize });
  }

  public static removeToken(id: number) {
    return Request.delete(`${urlBase}/open-api-token`, { id });
  }
}

File: jarboot-server/src/main/java/io/github/majianzheng/jarboot/service/UserService.java
package io.github.majianzheng.jarboot.service;

import io.github.majianzheng.jarboot.common.pojo.PagedList;
import io.github.majianzheng.jarboot.entity.User;

/**
 * @author majianzheng
 */
public interface UserService {
    /**
     * 创建用户
     * @param username 用户名
     * @param fullName 姓名
     * @param password 密码
     * @param roles 角色
     * @param userDir 用户关联目录
     * @param avatar 头像
     */
    void createUser(String username, String fullName, String password, String roles, String userDir, String avatar);

    /**
     * 删除用户
     * @param id 用户ID
     */
    void deleteUser(Long id);

    /**
     * 更新密码
     * @param loginUsername 当前登陆的用户
     * @param username 用户
     * @param oldPassword 旧密码
     * @param password 密码
     */
    void updateUserPassword(String loginUsername, String username, String oldPassword, String password);

    /**
     * 更新用户信息
     * @param username 用户
     * @param fullName 姓名
     * @param roles 角色
     * @param userDir 用户目录
     * @param avatar 头像
     */
    void updateUser(String username, String fullName, String roles, String userDir, String avatar);

    /**
     * 根据用户名获取用户信息
     * @param username 用户名
     * @return 用户信息
     */
    User findUserByUsername(String username);

    /**
     * 获取用户列表
     * @param username 用户名
     * @param role 角色
     * @param pageNo 开始页
     * @param pageSize 页大小
     * @return 用户列表
     */
    PagedList<User> getUsers(String username, String role, int pageNo, int pageSize);

    /**
     * 获取头像
     * @param username 用户名
     * @return 头像
     */
    String getAvatar(String username);
}
